ydb/library/yql/providers/generic/provider/yql_generic_list.cpp

```diff
@@ -0,0 +1,281 @@
+#include <yql/essentials/minikql/mkql_type_ops.h>
+#include <yql/essentials/providers/common/provider/yql_provider_names.h>
+#include <yql/essentials/providers/common/structured_token/yql_token_builder.h>
+#include <ydb/library/yql/providers/generic/connector/libcpp/client.h>
+#include <ydb/library/yql/providers/generic/connector/libcpp/error.h>
+#include <ydb/library/yql/providers/generic/expr_nodes/yql_generic_expr_nodes.h>
+#include <yql/essentials/utils/log/log.h>
+#include <ydb/core/external_sources/iceberg_fields.h>
+#include <ydb/library/yql/providers/generic/provider/yql_generic_predicate_pushdown.h>
+#include <ydb/library/yql/dq/expr_nodes/dq_expr_nodes.h>
+#include <ydb/library/yql/providers/generic/connector/libcpp/utils.h>
+
+namespace NYql {
+
+using namespace NNodes;
+using namespace NKikimr;
+using namespace NKikimr::NMiniKQL;
+
+TGenericListTransformer::TGenericListTransformer(TGenericState::TPtr state) 
+    : State_(std::move(state))
+{ }
+
+void FillColumns(NConnector::NApi::TSelect& select, const TGenReadTable& reader, NYql::NConnector::NApi::TSchema schema) {
+    auto items = select.mutable_what()->mutable_items();
+    auto columns = reader.Columns().Ptr();
+
+    if (!columns->IsList()) {
+        const auto rowType = reader.Ref()
+            .GetTypeAnn()
+                ->Cast<TTupleExprType>()
+                ->GetItems()
+            .back()
+                ->Cast<TListExprType>()
+                ->GetItemType();
+
+        const auto& exp = rowType->Cast<TStructExprType>()->GetItems();
+
+        for(auto item : exp) {
+            auto column = items->Add()->mutable_column();
+            column->mutable_name()->assign(item->GetName());
+            auto type = NConnector::GetColumnTypeByName(schema, TString(item->GetName()));
+            *column->mutable_type() = type;
+        }
+
+        return;
+    }
+
+    for (size_t i = 0; i < columns->ChildrenSize(); i++) {
+        auto cc = columns->Child(i);
+        Y_ENSURE(cc->IsAtom());
+
+        auto column = items->Add()->mutable_column();
+        auto columnName = TString(cc->Content());
+        column->mutable_name()->assign(columnName);
+
+        // assign column type
+        auto type = NConnector::GetColumnTypeByName(schema, columnName);
+        *column->mutable_type() = type;
+    }
+
+}
+
+IGraphTransformer::TStatus TGenericListTransformer::DoTransform(TExprNode::TPtr input,
+                                                                TExprNode::TPtr& output,
+                                                                TExprContext& ctx) {
+    output = input;
+
+    std::unordered_map<TListResponseMap::key_type, NConnector::NApi::TSelect, TListResponseMap::hasher> pendingTables;
+
+    const auto& reads = FindNodes(input, [&](const TExprNode::TPtr& node) {
+        if (const auto maybeRead = TMaybeNode<TGenReadTable>(node)) {
+            return maybeRead.Cast().DataSource().Category().Value() == GenericProviderName;
```

Comment:
У `TGenReadTable` на сколько я знаю обязательно должен быть провайдер `GenericProviderName`, можно просто вернуть `true` (или дополнительно проверить):

```suggestion
            Y_ENSURE(maybeRead.Cast().DataSource().Category().Value() == GenericProviderName);
            return true;
```